llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -7047,6 +7047,13 @@ static bool planContainsAdditionalSimplifications(VPlan &Plan,
       if (isa<VPPartialReductionRecipe>(&R))
         return true;
 
+      // VPBlendRecipes are converted to selects and may have been simplified.
+      using namespace VPlanPatternMatch;
+      if (match(&R, m_VPInstruction<Instruction::Select>(
```

Comment:
Why is this needed? This effectively disables cost checking for any plans with blends? If the blend gets simplified away, the regular checks should already catch this?

Reply:
Argh I see, so we create multiple selects with the same underlying instruction and this is causing the issue.

As lowering blends to selects becomes mandatory, it would probably be better to do it in convertToConcreteRecipes, at least to start with, removing the cost-model issue?

Reply:
We could do it in convertToConcreteRecipes but then we don't get any of the benefits of the simplifications that occur on selects, both in `VPlanTransforms::simplifyRecipes` and in `optimizeMaskToEVL`. 

Although I'm happy to at least start in convertToConcreteRecipes if you think it'll be ok to move the transform to earlier in a follow up patch?

Reply:
Yes that would probably be good as a first step, to start with removing VPBlendRecipe::execute.